src/server/game/Spells/SpellMgr.cpp

```diff
@@ -4650,6 +4650,153 @@ void SpellMgr::LoadSpellInfoCorrections()
 
     // ENDOF ANTORUS THE BURNING THRONE SPELLS
 
+    //
+    // SANCTUM OF DOMINATION
+    //
+
+    // Barbed Shot
+    ApplySpellFix({ 347807 }, [](SpellInfo* spellInfo)
+    {
+        spellInfo->AuraInterruptFlags |= SpellAuraInterruptFlags::LeavingCombat;
+        spellInfo->AuraInterruptFlags2 |= SpellAuraInterruptFlags2::EndOfEncounter;
+    });
+
+    // Ranger's Heartseeker
+    ApplySpellFix({ 352663 }, [](SpellInfo* spellInfo)
+    {
+        ApplySpellEffectFix(spellInfo, EFFECT_1, [](SpellEffectInfo* spellEffectInfo)
+        {
+            spellEffectInfo->ApplyAuraName = SPELL_AURA_PERIODIC_TRIGGER_SPELL;
+            spellEffectInfo->ApplyAuraPeriod = 180; // Sniffed value
+        });
+    });
+
+    // Domination Chain (Player)
+    ApplySpellFix({ 349451 }, [](SpellInfo* spellInfo)
+    {
+        spellInfo->AttributesEx2 |= SPELL_ATTR2_CAN_TARGET_NOT_IN_LOS;
+    });
+
+    // Wailing Arrow (Missile)
+    ApplySpellFix({ 348055 }, [](SpellInfo* spellInfo)
+    {
+        ApplySpellEffectFix(spellInfo, EFFECT_0, [](SpellEffectInfo* spellEffectInfo)
+        {
+            spellEffectInfo->BasePoints = 348056;
+        });
+    });
+
+    // Wailing Arrow (AoE)
+    ApplySpellFix({ 348056 }, [](SpellInfo* spellInfo)
+    {
+        ApplySpellEffectFix(spellInfo, EFFECT_0, [](SpellEffectInfo* spellEffectInfo)
+        {
+            spellEffectInfo->TriggerSpell = 357617;
+        });
+
+        ApplySpellEffectFix(spellInfo, EFFECT_1, [](SpellEffectInfo* spellEffectInfo)
+        {
+            spellEffectInfo->TriggerSpell = 357618;
+        });
+    });
+
+    // Veil of Darkness (Phase 1)
+    ApplySpellFix({ 347726 }, [](SpellInfo* spellInfo)
+    {
+        ApplySpellEffectFix(spellInfo, EFFECT_0, [](SpellEffectInfo* spellEffectInfo)
+        {
+            spellEffectInfo->Effect = SPELL_EFFECT_TRIGGER_SPELL;
+            spellEffectInfo->TriggerSpell = 347704;
+        });
+
+        ApplySpellEffectFix(spellInfo, EFFECT_1, [](SpellEffectInfo* spellEffectInfo)
+        {
+            spellEffectInfo->Effect = SPELL_EFFECT_TRIGGER_SPELL;
+            spellEffectInfo->TriggerSpell = 350777;
+        });
+    });
+
+    // Rive Marker
+    ApplySpellFix({ 353419 }, [](SpellInfo* spellInfo)
+    {
+        spellInfo->DurationEntry = sSpellDurationStore.LookupEntry(28); // 5s
```

Comment:
Can you post a sniff where the duration is 5 secs?

Reply:
The duration seems to be dynamic, but the longer it can last is 5s. Since I have no way to modify the AT's duration, I simply set it to the maximum so that I might call the AT's removefromworld on demand later.

Reply:
It's really good to leave a comment in code about not obvious things like this one

Reply:
Can you provide us some packets with the different durations?

Reply:
I can't because the duration is always set to 5s, tho the remove may be called sooner. It also depends on difficulty, e.g.: on Mythic, these markers only appear for 1s right before Sylvanas spawns the spike AT to reduce the reaction time window players have (when the AT reaches splinepoint 3). However, on the rest of the difficulties, it's spawned on a dynamic timer, and it's always removed 500ms after the AT spike is reaches splinepoint 3.